zeebe/journal/src/main/java/io/camunda/zeebe/journal/JournalInfoProvider.java

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.journal;
+
+import java.util.concurrent.CompletableFuture;
+
+public interface JournalInfoProvider {
```

Comment:
:wrench: If the only implementation of the interface is the `RaftServer`, this is perhaps better moved to the module holding that class, as it doesn't really have anything to do with journal. That likely means removing the dependency on the `journal` module from the `backup` module?

Reply:
In hindsight I put it here because I though the journal would implement it, but later realized that journal does not use future at all and so I just implemented in raft module.
I'll move it. This means that `zeebe-backup` will depend on `zeebe-atomix-cluster` which currently does not.

Reply:
Hm, the other option is putting it in a module that both `zeebe-atomix-cluster` and `zeebe-backup` share as dependency, and where it would also make sense. But I honestly don't know of any. 

A third option is having it as an interface purely in `zeebe-backup`, as a way to specify its dependencies, and have the place where we wire it together provide an adapter. This fully decouples both modules, and makes the integration module responsible. I think that's also fine, imo, esp. since the interface is (so far) a special purpose case for the backups :shrug:

Reply:
I went for the third option as it's the cleanest as you suggested.